Compiler/test2/cc_bytecode_test_1.cpp

```diff
@@ -3558,3 +3558,62 @@ TEST_F(Bytecode1, Linenum02)
     EXPECT_EQ(stringssize, scrip.strings.size());
 }
 
+TEST_F(Bytecode1, CharArrayDecay01)
+{
+    // An array of characters can be a parameter of a variadic function
+
+    char const *inpl = "\
+        import void Display(const string message, ...); \n\
+        short dummy;                \n\
+        char c[100];                \n\
+        int game_start()            \n\
+        {                           \n\
+            Display(\"<%s>\", c);   \n\
+        }                           \n\
+        ";
+
+    int compile_result = cc_compile(inpl, kNoOptions, scrip, mh);
+    std::string err_msg = mh.GetError().Message;
+    ASSERT_STREQ("Ok", mh.HasError() ? err_msg.c_str() : "Ok");
+
+    // WriteOutput("CharArrayDecay01", scrip);
+    size_t const codesize = 34;
+    EXPECT_EQ(codesize, scrip.code.size());
+
+    int32_t code[] = {
+      36,    5,   38,    0,           36,    6,    6,    2,    // 7
+       2,    3,    2,    3,           34,    3,    6,    3,    // 15
+       0,   34,    3,   39,            2,    6,    3,    0,    // 23
+      33,    3,   35,    2,           36,    7,    6,    3,    // 31
+       0,    5,  -999
+    };
+    CompareCode(&scrip, codesize, code);
+
+    size_t const numfixups = 3;
+    EXPECT_EQ(numfixups, scrip.fixups.size());
+
+    int32_t fixups[] = {
+       8,   16,   23,  -999
+    };
+    char fixuptypes[] = {
+      1,   3,   4,  '\0'
+    };
+    CompareFixups(&scrip, numfixups, fixups, fixuptypes);
+
+    int const numimports = 1;
+    std::string imports[] = {
+    "Display",      "[[SENTINEL]]"
```

Comment:
The import here will be registered as "Display^101" now, after the fixed #2300 .
"101" means a variadic function with 1 explicit parameter.

Reply:
This is an issue!

The bytecode googletests call the parser as-is and then check whether the compiler outputs are exactly what the googletest expects. So the googletest should either have the correct string `"Display^101"` or fail when the compiler calculates `"Display"` erroneously. 

I hope that I haven't mucked up something by mistake. I'm looking into why the function isn't registering correctly at the moment.

Reply:
Maybe there's a misunderstanding, the compiler does register this function correctly as "Display^101" (after #2300).
It's this new google test that has it wrong. Simply changing to "Display^101" here would solve the problem.

Reply:
Just did that (fix the googletest)

Reply:
I'm confused.

I've run the following program as a Googletest:
(Named, `CharArrayAsString01` at the end of file `cc_bytecode_test_1.cpp`: 

```
int printme1(const string text)
{
    Display(text);
}

int game_start()
{
    char c[10];
    c[0] = 'A';

    printme1("ab");
    printme1(c);
}

import void Display(const string, ...);
```

(Note that the compiler emits a `zeromem` for `c`, so this array is guaranteed to be `\0` terminated.)

This is what the compiler actually makes of the call of `Display():`
```
   4:   sourceline   3
   6:   load.sp.offs   8      ; move MAR to the location of 'text'
   8:   mov     mar,   ax    ; load AX with that location
  11:   farpush   ax
  13:   setfuncargs   1
  15:   movl    ax,   0    ; import number of 'Display()'
  18:   farcall   ax
  20:   farsubsp   1
```
That looks okay to me.

This is what the compiler actually makes of the calls of `printme1()`:
```
  50:   sourceline   11
  52:   movl    ax,   0    ; constant text block start of 'ab'
  55:   push    ax
  57:   movl    ax,   0    ; address of 'printme()' 
  60:   call    ax
  62:   subi    sp,   4
  65:   sourceline   12
  67:   load.sp.offs   10    ; move MAR to the location of 'c'
  69:   mov     mar,   ax    ; load AX with this location
  72:   push    ax
  74:   movl    ax,   0
  77:   call    ax
  79:   subi    sp,   4
```
Both calls look OK to me, too. 

At which point would the Engine ‘eat’ the contents of `c`?